src/dialogs/copyPasteOwnership.tsx

```diff
@@ -0,0 +1,185 @@
+        const split = owner.split(':');
+        let key = owner;
+        if (split.length === 2 && split[0] === split[1]) {
+            key = split[0];
+        }
+        if (owner === firstOwnerStr) {
+            key = `${key} ${_("(original owner)")}`;
+        }
+
+        map.set(key, owner);
+    });
+
+    return map;
+}
+
+const CopyPasteAsOwnerModal = ({
+    clipboard,
+    dialogResult,
+    path,
+} : {
+    clipboard: ClipboardInfo,
+    dialogResult: DialogResult<void>,
+    path: string,
+}) => {
+    const [currentUser, setCurrentUser] = useState<cockpit.UserInfo| undefined>();
+    const [selectedOwner, setSelectedOwner] = useState<string | undefined>();
+    const { cwdInfo, addAlert } = useFilesContext();
+
+    useEffect(() => {
+        cockpit.user().then(user => setCurrentUser(user));
+    }, []);
+
+    let candidatesMap: Map<string, string> = new Map();
+    if (superuser.allowed && currentUser && cwdInfo) {
+        candidatesMap = makeCandidatesMap(currentUser, cwdInfo, clipboard);
+        if (selectedOwner === undefined) {
+            setSelectedOwner(candidatesMap.keys().next().value);
+        }
+    }
+
+    if (selectedOwner === undefined) {
+        return;
+    }
+
+    const selectedVal = candidatesMap.get(selectedOwner);
+    cockpit.assert(selectedVal !== undefined, "New file ownership undefined");
+
+    const modalFooter = (
+        <>
+            <Button
+              variant="primary"
+              onClick={() => {
+                  pasteAsOwner(clipboard, path, selectedVal, addAlert);
+                  dialogResult.resolve();
+              }}
+            >
+                {_("Paste")}
+            </Button>
+            <Button variant="link" onClick={() => dialogResult.resolve()}>{_("Cancel")}</Button>
+        </>
+    );
+
+    return (
+        <Modal
+          id="paste-owner-modal"
+          position="top"
+          title={_("Paste as owner")}
+          isOpen
+          onClose={() => dialogResult.resolve()}
+          variant={ModalVariant.small}
+          footer={modalFooter}
+        >
+            <Form isHorizontal>
+                <TextContent>
+                    <Text>
+                        {_(`Files being pasted have a different owner. By default,
+                            ownership will be changed to match the destination directory.`)}
+                    </Text>
+                </TextContent>
+                <FormGroup fieldId="paste-as-owner" label={_("New owner")}>
+                    <FormSelect
+                      id='paste-owner-select'
+                      value={selectedOwner}
+                      onChange={(_ev, val) => setSelectedOwner(val)}
+                    >
+                        {[...candidatesMap.keys()].map(user =>
+                            <FormSelectOption
+                              key={user}
+                              value={user}
+                              label={user}
+                            />)}
```

Comment:
I find that overly excessive line splitting, but matter of taste -- no veto.

Reply:
blame eslint config :(